lib/logstash/inputs/jdbc.rb

```diff
@@ -232,4 +264,24 @@ def update_state_file
     end
   end
 
+  private
+
+  def enable_encoding?
+    !@charset.nil? || !@columns_charset.empty?
```

Comment:
small perf improvement suggestion: since this method is called on every row, we could memoize the `!@charset.nil? || !@columns_charset.empty?` expression since it will never change